### PR TITLE
Add streams return type  & new connector operations

### PR DESCRIPTION
**Pull request by SanduDS**
## Approach

1. Added streams for search returns
2. Fix account record field issues
3. Add contact search operation
4. Added the facilities to search by transaction types eg: invoices/salesorders etc
5. Added test cases

## Automation tests
 - Unit tests 
   > Done
 - Integration tests
   > Done

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: Invoice.bal [hunk 1]
```diff
@@ -151,41 +150,32 @@ isolated function getTranscationSearchRequestBody(SearchElement[] SearchElements
...
-    
+    return recordRefs;
+}
+
+isolated function getTransactionsNextPageResult(http:Response response) returns @tainted record {|RecordRef[] recordRefs; SearchResultStatus status;|}|error {
+    SearchResultStatus resultStatus = check getXMLRecordListFromSearchResult(response);
+    return {recordRefs :check getTransactionsFromSearchResults(resultStatus.recordList), status: resultStatus};
+}
+
+isolated function getTransactionSearchResult(http:Response response, http:Client httpClient, NetSuiteConfiguration config) returns @tainted stream<RecordRef, error>|error {
```
Comment (LakshanSS) [inline Invoice.bal:176]: We shouldn't exceed 120 characters. Please check all the places :)

File: client.bal [hunk 2]
```diff
@@ -231,46 +231,60 @@ public client class Client {
...
-    isolated remote function searchTransactionRecord(@display{label: "Search elements"} SearchElement[] searchElements) 
-                                                     returns @tainted @display{label: "Response"} RecordList|error {
+    @display{label: "Search Transactions"}
+    isolated remote function searchTransactionRecords(@display{label: "Search Elements"} SearchElement[] searchElements) 
+                                                     returns @tainted @display{label: "Response"} stream<RecordRef, 
+                                                     error>|error {
         xml payload = check buildTransactionSearchPayload(self.config, searchElements);
         http:Response response = check sendRequest(self.basicClient, SEARCH_SOAP_ACTION, payload);
-        return getTransactionSearchResult(response);
+        return getTransactionSearchResult(response,self.basicClient, self.config);
```
Comment (LakshanSS) [inline client.bal:253]: Need a space after `response,`

File: streams.bal [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            self.index += 1;
+            return singleRecord;
+        }
+    }
+
+    isolated function fetchCustomers() returns @tainted Customer[]|error {
+        xml payload = check buildSearchMoreWithIdPayload(self.config, self.currentPage + 1, self.searchId);
+        http:Response response = check sendRequest(self.httpClient, SEARCH_MORE_WITH_ID, payload);
+        record {|Customer[] customers; SearchResultStatus status;|} newPage = check getCustomersNextPageResult(response);
+        self.currentPage=newPage.status.pageIndex;
```
Comment (LakshanSS) [inline streams.bal:57]: Space needed before and after `=`

File: streams.bal [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+            return singleRecord;
+        }
+    }
+
+    isolated function fetchTransactions() returns @tainted Contact[]|error {
+        log:printInfo("Fetching next record set!");
+        xml payload = check buildSearchMoreWithIdPayload(self.config, self.currentPage + 1, self.searchId);
+        http:Response response = check sendRequest(self.httpClient, SEARCH_MORE_WITH_ID, payload);
+        record {|Contact[] contacts; SearchResultStatus status;|} newPage = check getContactsNextPageResult(response);
+        self.currentPage=newPage.status.pageIndex;
```
Comment (LakshanSS) [inline streams.bal:186]: Formatting issue

File: Package.md [hunk 5]
```diff
@@ -31,12 +31,12 @@ and search operations to perform business processing on NetSuite records and to
```
Comment (abeykoon) [inline Package.md:31]: Better add search types we support to the document.

**Review by abeykoon — APPROVED** [review]

**Pull request merged**